### PR TITLE
Skip pre-compressing assets for non-production images

### LINKED ISSUE #2478: Speed up asset compression for PR builds
Still needs to happen in some fashion so the uncompressed and compressed files have the same content, otherwise nginx will serve the previous release's gzipped assets. Maybe skip PNGs and run just one iteration for text files.

### PR DESCRIPTION
**Pull request by kberzinch**
Fixes #2478

### CODE REVIEW

**Review by JoshOldenburg — DISMISSED** [review]

File: .nomad/apiary.nomad [hunk 1]
```diff
@@ -222,7 +227,7 @@ EOF
         }
 
         meta {
-          nginx-config = trimspace(trimsuffix(trimspace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(trimspace(file("conf/nginx.conf")),"server\\s{\\s",""),"server_name\\s\\S+;",""),"root\\s\\S+;",""),"listen\\s.+;",""),"#.+\\n",""),";\\s+",";"),"{\\s+","{"),"\\s+"," ")),"}"))
+          nginx-config = var.precompressed_assets ? trimspace(trimsuffix(trimspace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(trimspace(file("conf/nginx.conf")),"server\\s{\\s",""),"server_name\\s\\S+;",""),"root\\s\\S+;",""),"listen\\s.+;",""),"#.+\\n",""),";\\s+",";"),"{\\s+","{"),"\\s+"," ")),"}")) : trimspace(trimsuffix(trimspace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(trimspace(file("conf/nginx.conf")),"server\\s{\\s",""),"gzip_static\\s\\S+;",""),"server_name\\s\\S+;",""),"root\\s\\S+;",""),"listen\\s.+;",""),"#.+\\n",""),";\\s+",";"),"{\\s+","{"),"\\s+"," ")),"}"))
```
Comment (evan10s) [inline .nomad/apiary.nomad:230]: Maybe split the ternary into variables for the different values? This was already mush but it's a lot more complicated now.

**Review by evan10s — CHANGES_REQUESTED** [review]
Lgtm besides some of the really complex Nginx mush in Nomad

File: .nomad/apiary.nomad (hunk 1 above)
Comment (kberzinch) [reply .nomad/apiary.nomad:230]: Mmmm, yeah I guess I can make locals...let me see

Comment (kberzinch) [reply .nomad/apiary.nomad]: Updated

**Review by evan10s — APPROVED** [review]

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**